Sources/Controllers/ConfigureScreen/DefaultMapButtonsViewController.swift

```diff
@@ -0,0 +1,179 @@
+        compassRow.accessibilityLabel = compassRow.title
+        compassRow.accessibilityValue = compassRow.descr
+        compassRow.iconTintColor = compassButtonState.isEnabled() ? iconTintColor : UIColor.iconColorDefault
+        compassRow.icon = compassButtonState.getIcon()
+        
+        let map3dModeRow = buttonsSection.createNewRow()
+        map3dModeRow.key = "map3DMode"
+        map3dModeRow.cellType = OAValueTableViewCell.reuseIdentifier
+        map3dModeRow.title = map3DButtonState.getName()
+        map3dModeRow.descr = getDescription(map3DButtonState)
+        map3dModeRow.accessibilityLabel = map3dModeRow.title
+        map3dModeRow.accessibilityValue = map3dModeRow.descr
+        map3dModeRow.iconTintColor = map3DButtonState.isEnabled() ? iconTintColor : UIColor.iconColorDefault
+        map3dModeRow.icon = map3DButtonState.getIcon()
+    }
+    
+    override func getRow(_ indexPath: IndexPath) -> UITableViewCell? {
+        let item = tableData.item(for: indexPath)
+        if item.cellType == OAValueTableViewCell.reuseIdentifier {
+            let cell = tableView.dequeueReusableCell(withIdentifier: OAValueTableViewCell.reuseIdentifier, for: indexPath) as! OAValueTableViewCell
+            cell.accessoryType = .disclosureIndicator
+            cell.descriptionVisibility(false)
+            cell.valueLabel.text = item.descr
+            cell.leftIconView.image = item.icon
+            cell.leftIconView.tintColor = item.iconTintColor
+            cell.titleLabel.text = item.title
+            cell.accessibilityLabel = item.accessibilityLabel
+            cell.accessibilityValue = item.accessibilityValue
+            return cell
+        }
+        return nil
+    }
+    
+    override func onRowSelected(_ indexPath: IndexPath) {
+        let data = tableData.item(for: indexPath)
+        if data.key == "compass" {
+            let vc = CompassVisibilityViewController()
+            vc.delegate = self
+            showMediumSheetViewController(vc, isLargeAvailable: false)
+        } else if data.key == "map3DMode" {
+            let vc = Map3dModeButtonVisibilityViewController()
+            vc.delegate = self
+            showMediumSheetViewController(vc, isLargeAvailable: false)
+        }
+    }
+    
+    // MARK: Additions
+    
+    private func onSettingsChanged() {
+        reloadDataWith(animated: true, completion: nil)
+        delegate?.onButtonsChanged()
+    }
+    
+    private func getDescription(_ buttonState: MapButtonState) -> String {
+        switch buttonState {
+        case let map3DButtonState as Map3DButtonState:
+            return map3DButtonState.getVisibility().title
+        case let compassButtonState as CompassButtonState:
+            return compassButtonState.getVisibility().title
+        default:
+            return ""
+        }
+    }
+}
+
+extension DefaultMapButtonsViewController: WidgetStateDelegate, OACopyProfileBottomSheetDelegate {
```

Comment:
Good practice is to use each delegate in its own extension. // MARK: WidgetStateDelegate is placed before the extension.